plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,648 @@
+This proves the exitability of the transaction.
+2. The user may present the transaction along signatures the user claims to be valid.
+The contract can validate that the exiting transaction is correctly formed.
+Another user can challenge one of these signatures by presenting some transaction that created an input such that the true input owner did not sign the signature.
+In this case, the exit would be blocked entirely and the challenging user would receive `exit bond`.
+
+Option (1) checks that a transaction is exitable when the exit is started.
+This has lower communication cost and complexity but higher up-front gas cost.
+This option also ensures that only a single exit on any given transaction can exist at any point in time.
+Option (2) allows a user to assert that a transaction is exitable, but leaves the proof to a challenge-response game.
+This is cheaper up-front but adds complexity.
+This option must permit multiple exits on the same transaction, as some exits may provide invalid signatures.
+
+These are not the only possible mechanisms that prove a transaction is exitable.
+There may be further ways to optimize these two options.
+
+We still need to provide a deterministic ordering of exits by some priority.
+MoreVP exits are given a priority based on the position in the Plasma chain of the most recently included (youngest) input to that transaction.
+Unlike the MVP protocol, we give each input and output to a transaction the same priority.
+This should be implemented by inserting a single “exit” object into a priority queue of exits and tracking a list of inputs or outputs to be paid out once the exit is processed.
+
+
+#### Proving Canonicity
+
+Whether unspent inputs or unspent outputs are paid out in an exit depends on the canonicity of the referenced transaction.
+Unfortunately it’s too expensive to directly prove that a transaction is or is not canonical.
+Instead, we assume that the referenced transaction is canonical by default and allow a series of challenges and responses to determine the true canonicity of the transaction.
+
+The process of determining canonicity involves a challenge-response game.
+In the first period of the exit, any user may reveal a competing transaction that potentially makes the exiting transaction non-canonical.
+This competing transaction must be exitable, must share an input with the exiting transaction, and must be included in the Plasma chain.
+Multiple competing transactions can be revealed during this period, but only the oldest presented transaction is considered for the purposes of a response.
+
+If any transactions have been presented during the first period, any other user can respond to the challenge by proving that the exiting transaction is actually included in the chain before the oldest presented competing transaction.
```

Comment:
I'd leave a `TODO`/placeholder for a diagram to depict the state machine of a single tx being inflight exited, in terms of determining canonicity. Somethin along the lines of `canonical -> non-canonical -> canonical` where the arrows are conditional on stuff being submited to the root chain/revealed